### PR TITLE
refactor(core): removal of deserialization logs for routing_metadata

### PR DESCRIPTION
**Pull request by prajjwalkumar17**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Removal of metadata_parsing logs, as it was being spammed a lot of times. Moreover soon a method for proper deserialization by either String match or conversion of the hashMap to typed will be done.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

We are just removing logs in this PR.
No extra code/flow changes.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router/src/core/payments/routing.rs [hunk 1]
```diff
@@ -115,10 +115,7 @@ pub fn make_dsl_input_for_payouts(
         .transpose()
         .change_context(errors::RoutingError::MetadataParsingError)
         .attach_printable("Unable to parse routing_parameters from metadata of payouts")
-        .unwrap_or_else(|err| {
-            logger::error!(error=?err);
-            None
-        });
+        .unwrap_or_else(|err| None);
```
Comment (ArjunKarthik) [inline crates/router/src/core/payments/routing.rs:118]: err is used, replace this with _

File: crates/router/src/core/payments/routing.rs [hunk 2]
```diff
@@ -937,10 +931,7 @@ pub async fn perform_session_flow_routing(
         .transpose()
         .change_context(errors::RoutingError::MetadataParsingError)
         .attach_printable("Unable to parse routing_parameters from metadata of payment_intent")
-        .unwrap_or_else(|err| {
-            logger::error!(?err);
-            None
-        });
+        .unwrap_or_else(|err| None);
```
Comment (ArjunKarthik) [inline crates/router/src/core/payments/routing.rs:934]: same here

File: crates/router/src/core/payments/routing.rs [hunk 3]
```diff
@@ -1166,10 +1157,7 @@ pub fn make_dsl_input_for_surcharge(
         .transpose()
         .change_context(errors::RoutingError::MetadataParsingError)
         .attach_printable("Unable to parse routing_parameters from metadata of payment_intent")
-        .unwrap_or_else(|err| {
-            logger::error!(error=?err);
-            None
-        });
+        .unwrap_or_else(|err| None);
```
Comment (ArjunKarthik) [inline crates/router/src/core/payments/routing.rs:1160]: same here

**Review by ArjunKarthik — APPROVED** [review]
LGTM

**Review by Aprabhat19 — APPROVED** [review]
Since the metadata based rules are not being consumed anywhere on the prod, this PR LGTM

**Pull request merged**